tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html

```diff
@@ -3941,16 +3979,31 @@ <h2>Create a distance feature</h2>
         // Convert the data into the format expected by vz-bar-chart.
         // Stack models and labels so that each (model,label) is a separate
         // series
+        const lines = {};
         const mapped = {};
+        const self = this;
         _.forEach(data, (modelValues, modelInd) => {
           const obj = _.forEach(modelValues,
             (pointsForSeries, labelKey) => {
+              // Sort bars by their inference values for display.
               mapped[this.formatChartKey(labelKey, modelInd, data.length)] =
-                _.map(pointsForSeries, point => convertPoint(point, origKey));
+                _.map(pointsForSeries, point => convertPoint(
+                  point, origKey)).sort((a, b) => b.y - a.y);
+              // For binary classifiers, also create lines to indicate the
+              // classification threshold for each model, with datapoints
+              // sorted by the inference values to match the bar order for
+              // the bar chart it will be displayed over.
+              if (this.isBinaryClassification_(
+                    this.modelType, this.multiClass)) {
+                lines[this.formatChartKey(labelKey, modelInd, data.length)] =
+                  _.map(pointsForSeries, point => convertThreshold(
+                    point, origKey,
+                    this.overallThresholds[modelInd].threshold)).sort(
+                      (a, b) => b.scalar - a.scalar);
```

Comment:
If above statement is correct then this may also need edits, also if y value is always constant (threshold), does the order matter?

Reply:
I do this to order the line segments in the same order as the bars, as the draw line has extra thickness when it needs to double-back to the left for a line segment. So having the series in the same order as the bars ensures this doesn't happen and the lines look uniform.